mitiq/ddd/insertion.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2022 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Tools to determine slack windows in circuits and to insert DDD sequences."""
+import numpy as np
+from mitiq.interface import convert_to_mitiq
+from mitiq import QPROGRAM
+
+
+def get_circuit_mask(circuit: QPROGRAM) -> np.ndarray:
+    """Given a circuit with n qubits and d moments returns a matrix
+    A with n rows and d columns. The matrix elements are A_{i,j} = 1 if
+    there is a gate acting on qubit i at moment j, while A_{i,j} = 0 otherwise.
+
+    Args:
+        circuit: Input circuit to mask with n qubits and d moments
+
+    Returns:
+        A mask matrix with n rows and d columns
+    """
+    mitiq_circuit, _ = convert_to_mitiq(circuit)
```

Comment:
I see that this function is compatible with any `QPROGRAM` but we test it only with Cirq circuits.
With other circuits there could be some problems (e.g. if a Braket gate is converted to a Cirq decomposition involving more gates the Cirq mask could be unexpected with respect to the input Braket circuit).
I see two options to proceed:

1. Explaining this potential unexpected behavior in the docstring. And maybe add a test with a different front end? 
2. Making this function private and compatible only with Cirq circuits : e.g.  _gate_circuit(circuit: cirq.Circuit): -> np.ndarray

Both options are good for me.

Reply:
Initially I went with (1) but looking over the RFC I think it will make the most sense to apply conversions with the decorating converter flow against `insert_ddd_sequences()`. So now it's (2).